packages/next/client/dev/fouc.ts

```diff
@@ -1,12 +1,24 @@
-// This wrapper function is used to avoid raising a Trusted Types violation.
-const safeSetTimeout = (callback: () => void) => setTimeout(callback)
+// This wrapper function is used to safely select the best available function
+// to schedule removal of the no-FOUC styles workaround. requestAnimationFrame
+// is the ideal choice, but when used in iframes, there are no guarantees that
+// the callback will actually be called, which could stall the promise returned
+// from displayContent.
+//
+// See: https://www.vector-logic.com/blog/posts/on-request-animation-frame-and-embedded-iframes
+const safeCallbackQueue = (callback: () => void) => {
+  if (window.requestAnimationFrame && window.self === window.top) {
```

Comment:
Even though the issue is with hidden iframes, I think it's safe to scope this check to iframes in general (given that this is a `dev` feature. Open to feedback here and can also detect if it's hidden or not here.